lib/service/service.go

```diff
@@ -3203,8 +3203,25 @@ func (process *TeleportProcess) initDiagnosticService() error {
 	logger.InfoContext(process.ExitContext(), "Starting diagnostic service.", "listen_address", process.Config.DiagnosticAddr.Addr)
 
 	process.RegisterFunc("diagnostic.service", func() error {
-		err := server.Serve(listener)
-		if err != nil && err != http.ErrServerClosed {
+		muxListener, err := multiplexer.New(multiplexer.Config{
```

Comment:
We should close `muxListener` either here in a `defer` or in `diagnostic.shutdown`.

Reply:
Done https://github.com/gravitational/teleport/pull/39497/commits/436d415b4fb9e4aa0f36538f295da7eefb8b1c11